packages/anasazi/tpetra/test/LOBPCG/cxx_main_solvertest.cpp

```diff
@@ -0,0 +1,468 @@
+// @HEADER
+// *****************************************************************************
+//                 Anasazi: Block Eigensolvers Package
+//
+// Copyright 2004 NTESS and the Anasazi contributors.
+// SPDX-License-Identifier: BSD-3-Clause
+// *****************************************************************************
+// @HEADER
+//
+//  This test is for the LOBPCG solver
+//
+#include "AnasaziConfigDefs.hpp"
+#include "AnasaziTypes.hpp"
+
+#include "AnasaziTpetraAdapter.hpp"
+#include "AnasaziBasicEigenproblem.hpp"
+#include "AnasaziBasicOutputManager.hpp"
+#include "AnasaziLOBPCGSolMgr.hpp"
+#include "AnasaziSVQBOrthoManager.hpp"
+#include "AnasaziBasicSort.hpp"
+#include "AnasaziStatusTestMaxIters.hpp"
+#include "AnasaziSolverUtils.hpp"
+#include "Teuchos_CommandLineProcessor.hpp"
+#include "Teuchos_SerialDenseMatrix.hpp"
+
+#include <Tpetra_Core.hpp>
+#include <Tpetra_CrsMatrix.hpp>
+
+#include <ModeLaplace1DQ1.hpp>
+
+using namespace Teuchos;
+using namespace Anasazi;
+
+typedef double                                      ST;
```

Comment:
ditto

Reply:
Test is only meaningful for double-precision due to generalized eigenvalue problem being generated.